src/plugin/svipc/sysvipcwrappers.cpp

```diff
@@ -222,13 +222,18 @@ int semctl(int semid, int semnum, int cmd, ...)
   va_start (arg, cmd);
   uarg = va_arg (arg, union semun);
   va_end (arg);
+  int ret = -1;
 
   DMTCP_PLUGIN_DISABLE_CKPT();
-  int realId = VIRTUAL_TO_REAL_SEM_ID(semid);
-  JASSERT(realId != -1);
-  int ret = _real_semctl(realId, semnum, cmd, uarg);
-  if (ret != -1) {
-    SysVSem::instance().on_semctl(semid, semnum, cmd, uarg);
+  if (cmd == SEM_INFO || cmd == IPC_INFO) {
```

Comment:
I suppose it's a matter of taste, but I would do:
```C
if (cmd == SEM_INFO || cmd == IPC_INFO) {
   return_real_semctl(semid, semnum, cmd, uarg);
}

DMTCP_PLUGIN_DISABLE_CKPT();
 ...
DMTCP_PLUGIN_ENABLE_CKPT();
```

This way, the exceptional case is inside an "if" statement, and the normal case is still at top level.